mllib/src/main/scala/org/apache/spark/ml/classification/MultilayerPerceptronClassifier.scala

```diff
@@ -65,7 +66,42 @@ private[ml] trait MultilayerPerceptronParams extends PredictorParams
   /** @group getParam */
   final def getBlockSize: Int = $(blockSize)
 
-  setDefault(maxIter -> 100, tol -> 1e-4, layers -> Array(1, 1), blockSize -> 128)
+  /**
+   * Optimizer setup.
+   * @group expertParam
+   */
+  final val optimizer: Param[String] = new Param[String](this, "optimizer",
+    " Allows setting the optimizer: minibatch gradient descent (GD) or LBFGS. " +
+      " The latter is recommended one. ",
+    ParamValidators.inArray[String](Array("GD", "LBFGS")))
```

Comment:
use lowercase letters: `gd` and `l-bfgs`. Note that we already used `l-bfgs` in `LinearRegression`, so we should do the same here.